x-pack/solutions/search/plugins/search_synonyms/public/components/synonym_sets/create_new_set_modal.tsx

```diff
@@ -54,31 +65,73 @@ export const CreateSynonymsSetModal = ({ onClose }: CreateSynonymsSetModalProps)
           component="form"
           onSubmit={(e) => {
             e.preventDefault();
-            createSynonymsSet({ synonymsSetId: name });
+            createSynonymsSet({ synonymsSetId: name, forceWrite });
           }}
         >
           <EuiFormRow
+            fullWidth
             label={i18n.translate('xpack.searchSynonyms.createSynonymsSetModal.nameLabel', {
               defaultMessage: 'Name',
             })}
             helpText={
-              !!rawName
+              !!rawName && !conflictError
                 ? i18n.translate('xpack.searchSynonyms.createSynonymsSetModal.nameHelpText', {
                     defaultMessage: 'Your synonyms set will be named: {name}',
                     values: { name },
                   })
                 : undefined
             }
+            isInvalid={conflictError}
+            error={
+              conflictError
+                ? i18n.translate('xpack.searchSynonyms.createSynonymsSetModal.nameErrorText', {
+                    defaultMessage: 'A synonym with id {name} already exists.',
+                    values: { name },
+                  })
+                : undefined
+            }
           >
             <EuiFieldText
               data-test-subj="searchSynonymsCreateSynonymsSetModalFieldText"
               value={rawName}
               onChange={(e) => {
                 setRawName(e.target.value);
                 setName(formatSynonymsSetName(e.target.value));
+                setConflictError(false);
+                setForceWrite(false);
               }}
             />
           </EuiFormRow>
+          {conflictError && (
+            <>
+              <EuiSpacer size="s" />
+              <EuiFormRow fullWidth>
+                <EuiText
+                  color="danger"
+                  data-test-subj="searchSynonymsCreateSynonymsSetModalError"
+                  size="s"
+                >
+                  <p>
+                    <FormattedMessage
+                      id="xpack.searchSynonyms.createSynonymsSetModal.error"
+                      defaultMessage="Creating a synonyms set with the same name will overwrite the existing set."
+                    />
+                  </p>
+                </EuiText>
+              </EuiFormRow>
+              <EuiFormRow fullWidth>
+                <EuiCheckbox
+                  id={overwriteId}
+                  data-test-subj="searchSynonymsCreateSynonymsSetModalForceWrite"
+                  label={i18n.translate('xpack.searchSynonyms.createSynonymsSetModal.forceWrite', {
+                    defaultMessage: 'I understand the risks. Overwrite the existing synonym set.',
```

Comment:
@leemthompo Thoughts on making this so it's just focused on the action (removing the 'I understand...')?  

I think it can just be `Overwrite the existing synonym set.` Risk makes it seem like there's a chance the old set won't be overwritten.